py/embedded-server/deephaven_server/__init__.py

```diff
@@ -14,3 +19,55 @@
 # distribution name from the package name.
 # https://docs.python.org/3/library/importlib.metadata.html#package-distributions
 __version__ = importlib.metadata.version('deephaven-server')
+
+
+def main():
+    """
+    Main entry point for the Deephaven embedded server application.
+    Accepts a command to start the application and the arguments to use.
+    Defaults to `start`, which starts the server.
+    """
+    parser = argparse.ArgumentParser(description="Start the Deephaven embedded server")
+    parser.add_argument("cmd", nargs="?", default="start", choices=["start"], help="Deephaven command to run. Defaults to start.")
+    parser.add_argument("-p", "--port", default=8080, type=int, help="The port to bind to. Defaults to 8080.")
+    parser.add_argument("--host", default="localhost", help="The host to bind to. Defaults to localhost.")
+    parser.add_argument("--key", default=None, help="The key to use. Uses a randomly generated key if not specified.")
+    parser.add_argument("--jvm-args", default=None, help="The JVM arguments to use.")
+    parser.add_argument("--dh-args", default=None, help="The Deephaven arguments to use.")
+    args = parser.parse_args()
+
+    if args.cmd == "start":
+        _start(args)
+    else:
+        parser.print_help()
+        sys.exit(1)
+
+def _start(args):
+    """
+    Start the Deephaven server with the given arguments.
+
+    Args:
+        args: The arguments parsed using argparse to use to start the server.
+    """
+    if args.key is None:
+        args.key = secrets.token_urlsafe(32)
+
+    jvm_args = [f"-Dauthentication.psk={args.key}"]
+    if args.jvm_args is not None:
+        jvm_args += args.jvm_args.split()
```

Comment:
Something we've talked about before is creating the config file if it doesn't exist and writing the PSK into it; this gives it some persistence. I think that's how jupyter does it?

Reply:
Jupyter doesn't seem to persist the token. If I run `jupyter lab` twice in a row, I get new tokens. I think that makes sense, it shouldn't persist unless you specify it.

That being said I'm not against it and Jupyter certainly doesn't have to be precedence.

Reply:
Okay; do you know if a fresh jupyter lab writes down any configuration files?

Reply:
Yes, it writes a couple of files it looks like:
- `/home/bender/.jupyter/lab/workspaces/default-37a8.jupyterlab-workspace`: Config for the users workspace, such as layout of the panels
- `/home/bender/.local/share/jupyter/runtime/jupyter_cookie_secret`: I'm not 100% sure what this cookie is for; it's not the token used for auth (that regenerates each time) and it just regenerates this if you delete the file. Searching for details on what that cookie is used for isn't turning up much (I checked the [jupyter](https://github.com/search?q=org%3Ajupyter%20jupyter_cookie_secret&type=code) and [jupyterlab](https://github.com/search?q=org%3Ajupyterlab+jupyter_cookie_secret&type=code) orgs on GitHub). There's some mention of it in forums indicating it'll fail if [permissions don't allow reading the file](https://github.com/jupyterhub/jupyterhub/issues/1787), but it will regenerate if the file doesn't exist.

So again, we can write a config, I don't mind writing the token to a file for persistence, but that doesn't seem to be what Jupyter is storing (and I'm actually not sure _what_ the secret cookie they're storing is anyway. Some sort of base64 string, but it doesn't seem to be stored in the browser as a cookie, doesn't seem to have a reference in any of the accessible Jupyter code that I've searched ... I don't know what it is?